src/coreComponents/functions/MultivariableTableFunctionKernels.hpp

```diff
@@ -0,0 +1,321 @@
+
+/**
+ * @brief interpolate operators at a given point
+ *
+ * @param[in] coordinates point coordinates
+ * @param[out] values interpolated operator values
+ * @param[out] derivatives derivatives of interpolated operators
+ */
+  GEOSX_HOST_DEVICE
+  void
+  interpolatePoint( real64 const *coordinates,
+                    real64 *values,
+                    real64 *derivatives ) const
+  {
+    globalIndex hypercubeIndex = 0;
+    real64 axisLows[NUM_DIMS];
+    real64 axisMults[NUM_DIMS];
+
+    for( int i = 0; i < NUM_DIMS; ++i )
+    {
+      integer axisIndex = getAxisIntervalIndexLowMult( coordinates[i],
+                                                       m_axisMinimums[i], m_axisMaximums[i],
+                                                       m_axisSteps[i], m_axisStepInvs[i], m_axisPoints[i],
+                                                       axisLows[i], axisMults[i] );
+      hypercubeIndex += axisIndex * m_axisHypercubeMults[i];
+    }
+
+    interpolatePointWithDerivatives( coordinates,
+                                     getHypercubeData( hypercubeIndex ),
+                                     &axisLows[0], &axisMults[0],
+                                     &m_axisStepInvs[0],
+                                     values,
+                                     derivatives );
+
+  }
+
+protected:
+
+  /**
+   * @brief Get pointer to hypercube data
+   *
+   * @param[in] hypercubeIndex
+   * @return pointer to hypercube data
+   */
+  GEOSX_HOST_DEVICE
+  inline
+  real64 const *
+  getHypercubeData( globalIndex const hypercubeIndex ) const
+  {
+    static constexpr integer NUM_VERTS = 1 << NUM_DIMS;
+    return &m_hypercubeData[hypercubeIndex * NUM_VERTS * NUM_OPS];
+  }
+
+  /**
+   * @brief Get the interval index, low and mult values for a given axis coordinate
+   *
+   * @param[in] axisCoordinate coordinate on a given axis
+   * @param[in] axisMin minimum value on a given axis
+   * @param[in] axisMax maximum value on a given axis
+   * @param[in] axisStep interval length for a given axis
+   * @param[in] axisStepInv inversion of the interval length for a given axis
+   * @param[in] axisPoints number of discretization points for a given axis
+   * @param[out] axisLow left coordinate of target axis interval
+   * @param[out] axisMult weight of the right coordinate of target axis interval
+   * @return integer target axis interval index
+   */
+  GEOSX_HOST_DEVICE
+  inline
+  integer
+  getAxisIntervalIndexLowMult( real64 const axisCoordinate,
+                               real64 const axisMin,
+                               real64 const axisMax,
+                               real64 const axisStep,
+                               real64 const axisStepInv,
+                               integer const axisPoints,
+                               real64 & axisLow,
+                               real64 & axisMult ) const
+  {
+    integer axisIntervalIndex = integer((axisCoordinate - axisMin) * axisStepInv );
```

Comment:
Understood, thanks for pointing it out! I can`t however apply the suggested change in this case (or fail to see how) because this variable might be changed after initialization, see a few lines below...

Reply:
Oops, my bad, I totally chose the wrong example! :disappointed: Please disregard the comment, clearly I have not read the code with enough attention.